github/github.go

```diff
@@ -889,6 +890,13 @@ func (c *Client) BareDo(ctx context.Context, req *http.Request) (*Response, erro
 			err = aerr
 		}
 
+		rateLimitError, ok := err.(*RateLimitError)
+		if ok && req.Context().Value(SleepUntilPrimaryRateLimitResetWhenRateLimited) != nil {
+			time.Sleep(time.Until(rateLimitError.Rate.Reset.Time))
+			// re-send the request
+			return c.BareDo(ctx, req)
```

Comment:
I think this is probably OK, but am wondering about two things:
* this recursively calls c.BareDo - at what point should it truly fail?
* The defered body close on line 876 doesn't actually happen until the recursion fully unwraps

Again, I think this is probably all fine.
Did you try testing this out, perchance?
I know the whole point is to prevent rate limit violations - but it might be nice to verify that this does what we want.

Reply:
I did test it and the retry works. Good point on the defer let me see if I can find a good solution for it. Another thing I'm considering is to only retry once